indico/modules/events/abstracts/operations.py

```diff
@@ -81,6 +83,28 @@ def delete_abstract(abstract, delete_contrib=False):
                            'Abstract "{}" has been deleted'.format(abstract.title), session.user)
 
 
+def merge_person_links(target_abstract, source_abstract):
+    """Add to `target_abstract` new `AbstractPersonLink`s whose `EventPerson`
+       exists in the `source_abstract` but is not yet in the `target_abstract`.
+
+       :param target_abstract: The target abstract (to which the links should then be added)
+       :param source_abstract: The source abstract
+    """
+    source_links = source_abstract.person_links
+    source_link_map = defaultdict(set)
+    for link in source_links:
+        source_link_map[link.person.id].add(link)
+    unique_persons = {link.person for link in source_links} - {link.person for link in target_abstract.person_links}
+
+    for person in unique_persons:
+        for source_link in source_link_map[person.id]:
+            link = AbstractPersonLink(person=person, author_type=source_link.author_type,
+                                      is_speaker=source_link.is_speaker)
+            target_abstract.person_links.append(link)
```

Comment:
I'd check if the target abstract had a custom order - if yes, max(position) + 1, otherwise I'd keep 0 (ie alphabetical order) even if some of the source abstracts had a custom order there.